lib/ec2metadata/__init__.py

```diff
@@ -68,11 +68,28 @@ def _add_mata_option(self, path):
             if item:
                 if item == 'public-keys/':
                     continue
-                if item not in options:
-                    if item[-1] != '/':
+                if item[-1] == '/':
+                    self._add_mata_option(path+item)
+                else:
+                    if item not in options:
                         self.meta_options_api_map[item] = path + item
                     else:
-                        self._add_mata_option(path+item)
+                        # Construct a new name for the option using the given
+                        # path as name addition
+                        option_name = item
+                        # Drop the empty entry, the path has a '/' at the end
+                        prefix_name_opts = path.split('/')[:-1]
+                        prefix_name_opts.reverse()
+                        cnt = 1
+                        while option_name in options:
+                            option_name = '-'.join(
+                                prefix_name_opts[:cnt] + [option_name]
```

Comment:
Not sure I follow logic without context but it seems (could be wrong) like we end up with two options:

- info
- ec2-identity-credentials-info

If there's any clash I would expect both names become verbose and I wonder if the dash in identitycredentials causes any trouble?

Expectation would be:

- iam-info
- ec2-identity-credentials-info

Do we split this back on dash? Then identity-credentials/ec2/info becomes identity/credentials/ec2/info?

Reply:
Well there is no way to predict if there is a clash. When the code process the first endpoint called `info` it does not know that there will be another.

If we expect the expansion of both names, which is a good suggesting that can be added but requires a bit more refactoring as we wold then have to remove the previously generated option name and insert the newly expanded name. But then if a third one comes along we are back to the short name, i.e. `info`. That can only be accounted for with an additional tracking list `duplicate_names` which in and off itself if not a problem.

But generally this is a good idea as the current approach of "first one wins the naming battle" is flaky in the sense that if IMDS changes the order in which things are presented than `--info` means one thing today and may mean a different thing tomorrow.

I'll do some refinement